### ISSUE TITLE
EnC: member moves across partial definitions might be classified as renamed and not handled correctly

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#51011]
**Version Used**: 
Version 16.9.0 Preview 4.0 [30920.304.main]

**Steps to Reproduce**:

Before edit

File 1:
```C#
partial class C
{
   void F1() {}
}
```

File 2:
```C#
partial class C
{
   void F2() {}
}
```

After edit:

File 1:
```C#
partial class C
{
   void F2() {}
}
```

File 2:
```C#
partial class C
{
   void F1() {}
}
```

See test `NestedPartialTypeInPartialType_RenameInsertDeleteAndInsertVirtual`

**Expected Behavior**:

Each document has semantic update of F1 and F2, respectively.

**Actual Behavior**:

"Can't rename" rude edit is reported.

**Comment by tmat** [comment]
Another scenario:

Before edit

File 1:

partial class C
{
   void F1() {}
}
File 2:

partial class C
{
   void F2() {}
}
After edit:

File 1:

partial class C
{
   void F1() {} // moved from file 2
   void F2() {}
}
File 2:

partial class C
{
   void F3() {} // added new method
}
